pkg/kv/import.go

```diff
@@ -127,16 +103,35 @@ func Import(ctx context.Context, reader io.Reader, store Store) error {
 	})
 	log.Info("Processing file")
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	entryChan := make(chan *Entry, entryQueueSize)
-	var g multierror.Group
+	workerErr := make([]error, importWorkers)
+	var wg sync.WaitGroup
+	wg.Add(importWorkers)
 	for i := 0; i < importWorkers; i++ {
-		g.Go(func() error {
-			return consumer(cctx, cancel, entryChan, store)
-		})
+		go func(idx int) {
+			defer wg.Done()
+			for e := range entryChan {
+				err := store.SetIf(ctx, e.PartitionKey, e.Key, e.Value, nil)
+				if err != nil {
+					workerErr[idx] = fmt.Errorf("import (partition key: %s, key: %s): %w", e.PartitionKey, e.Key, err)
+					return
+				}
+			}
+			if workerErr[idx] != nil {
+				cancel()
+			}
+		}(i)
 	}
-	g.Go(func() error {
-		return producer(cctx, cancel, log, entryChan, jd)
-	})
+	err := importReader(ctx, log, entryChan, jd)
+	close(entryChan)
+
+	wg.Wait()
 
-	return g.Wait().ErrorOrNil()
+	var merr *multierror.Error
+	merr = multierror.Append(merr, err)
+	merr = multierror.Append(merr, workerErr...)
+	return merr.ErrorOrNil()
```

Comment:
multiErr better?

Reply:
This is the same type used by `m.Go` - we can revert part of my change to use it and simplify it a bit more

Reply:
I was referring to the variable name